warehouse/integrations/deltalake/deltalake.go

```diff
@@ -1383,5 +1383,7 @@ func (*Deltalake) DeleteBy(context.Context, []string, warehouseutils.DeleteByPar
 // * the load table strategy is "append" mode
 // * the uploader says we can append
 func (d *Deltalake) ShouldAppend() bool {
-	return d.config.loadTableStrategy == appendMode && d.Uploader.CanAppend()
+	return d.config.loadTableStrategy == appendMode &&
+		d.Uploader.CanAppend() &&
+		!d.Warehouse.GetBoolDestinationConfig("enableDedup")
```

Comment:
@lvrach Same question here, does the load table strategy makes sense? Should we remove it?

Reply:
By default if nothing is set then we are in merge mode. Only if the customer opts-in we can go in append mode.

P.S.: `enableDedup` -> `enableAppend`.

Reply:
We need to document the behaviour of `upload.CanAppend()`. We can also move the `integration.warehouse.GetBoolDestinationConfig("enableAppend")` into the `upload.CanAppend()` method.